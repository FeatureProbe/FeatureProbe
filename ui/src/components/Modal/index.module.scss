.footer {
  margin-top: 20px;
  text-align: right;
}

.btn {
  margin-right: 8px !important;
}