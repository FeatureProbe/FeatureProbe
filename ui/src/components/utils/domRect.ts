export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}