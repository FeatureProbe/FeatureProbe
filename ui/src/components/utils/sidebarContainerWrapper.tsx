import { SidebarContainer } from 'layout/hooks';

export default SidebarContainer.Provider;
