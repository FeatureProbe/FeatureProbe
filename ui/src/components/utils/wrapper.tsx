import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
