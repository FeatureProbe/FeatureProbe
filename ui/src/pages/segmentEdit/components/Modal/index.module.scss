.modal-inner-box {
  min-height: 405px;
}

.modal-header {
  display: flex;
  justify-content: space-between;
  font-family: PingFangSC-Medium, sans-serif;
  font-size: 16px;
  color: #212529;
  height: 25px;
  margin-bottom: 9px;
}

.modal-header-icon {
  font-size: 16px !important;
  cursor: pointer;
  color: #495057;
}

.modal-info-circle {
  font-size: 17px;
  color: #0a70f5;
  margin-right: 9px;
}

.modal-tips {
  height: 36px;
  background: #EFF8FF;
  font-size: 14px;
  color: #042866;
  line-height: 36px;
  position: absolute;
  width: 100%;
  left: 0;
  padding: 0 16px;
}

.modal-content {
  margin-top: 52px;
  margin-bottom: 70px;
}

.modal-footer {
  display: flex;
  height: 43px;
  justify-content: flex-end;
  align-items: flex-end;
  border-top: 1px solid #ced4da;

  & > button {
    margin-left: 8px !important;
  }
}

.comment {
  margin: 24px -16px 0;
  padding: 0 24px;
  align-items: center;
}

.comment-title {
  margin-bottom: 8px;
}

.comment-content {
  height: 190px;
  overflow: auto;
}

.comment-input {
  :global {
    .ui.input {
      width: 100%;
    }

    .ui.input > input {
      height: 32px;
    }
  }
}

.diff-box {
  font-size: 14px;
  color: rgba(0, 0, 0, 0.6);
  margin: 0 -16px -4px;
}
