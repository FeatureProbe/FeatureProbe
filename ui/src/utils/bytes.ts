export function getBytes(str: string) {
  let bytes = str.length,
    i = 0;
  for (; i < bytes; i++) if (str.charCodeAt(i) > 255) bytes++;

  return bytes;
}

const MB = 1048576;
export const MAX_SIZE = MB;