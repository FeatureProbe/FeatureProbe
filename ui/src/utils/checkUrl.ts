function escapeStringRegexp(string: string): string {
  if (typeof string !== 'string') {
    throw new TypeError('Expected a string');
  }

  return string
    .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
    .replace(/-/g, '\\x2d');
}

export function matchUrl(matcher: string, originUrl: string, targetUrl: string): boolean {
  let href = '';
  let hash = '';
  let search = '';

  try {
    const fullUrl = new URL(targetUrl, '');
    href = fullUrl.href;
    hash = fullUrl.hash;
    search = fullUrl.search;
  } catch (error) {
    href = targetUrl;
  }

  let regex;
  let testUrl;

  switch (matcher) {
    case 'EXACT':
      testUrl = href;
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SIMPLE':
      testUrl = href.replace(hash, '').replace(search, '');
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SUBSTRING':
      testUrl = href.replace(search, '');
      regex = new RegExp('.*' + escapeStringRegexp(originUrl) + '.*$');
      break;
    case 'REGULAR':
      testUrl = href.replace(search, '');
      regex = new RegExp(originUrl);
      break;
    default:
      return false;
  }

  return regex.test(testUrl);
}