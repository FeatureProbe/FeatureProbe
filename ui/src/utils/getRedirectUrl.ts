import { getProjectList } from 'services/project';
import { IProject, IEnvironment } from 'interfaces/project';
import { IDictionary } from 'interfaces/targeting';
import { getFromDictionary } from 'services/dictionary';
import { LAST_SEEN } from 'constants/dictionary_keys';

const getInfo = async () => {
  const res =  await getFromDictionary<IDictionary>(LAST_SEEN);
  const { success, data } = res;
  if (success && data) {
    return JSON.parse(data.value);
  } else {
    return {};
  }
};

export const getRedirectUrl = async (defaultRedirectUrl: string) => {
  let redirectUrl = defaultRedirectUrl;
  const info = await getInfo();
  const { projectKey, environmentKey } = info;

  if (!!projectKey && !!environmentKey) {
    redirectUrl = `/${projectKey}/${environmentKey}/toggles`;
  } else {
    const res = await getProjectList<IProject[]>();
    const { success, data } = res;
    if (success && data) {
      if (data[0]) {
        const { key, environments } = data[0];
        let envKey = '';
        environments.some((environment: IEnvironment) => {
          if (environment.key) {
            envKey = environment.key;
            return true;
          }
          return environment;
        });
  
        if (key && envKey) {
          redirectUrl = `/${key}/${envKey}/toggles`;
        }
      } else {
        redirectUrl = '/projects';
      }
    }
  }

  return redirectUrl;
};