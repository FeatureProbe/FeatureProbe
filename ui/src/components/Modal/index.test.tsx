import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Modal from '.';
import { IntlWrapper } from 'components/utils/wrapper';

it('Modal snapshot', (done) => {
  (async () => {
    const { baseElement, rerender } = render(
      <Modal open={true}>
        <div>modal content</div>
      </Modal>,
      {
        wrapper: IntlWrapper,
      }
    );
    expect(baseElement).toMatchSnapshot();

    rerender(
      <Modal open={true} footer={<div>footer</div>}>
        <div>modal content</div>
      </Modal>
    );
    done();
  })();
});

test('click Modal button', (done) => {
  (async () => {
    const mockHandleCancel = jest.fn();
    const mockHandleConfirm = jest.fn();
    render(
      <Modal open={true} handleCancel={mockHandleCancel} handleConfirm={mockHandleConfirm}>
        <div>modal content</div>
      </Modal>,
      {
        wrapper: IntlWrapper,
      }
    );

    await userEvent.click(screen.getByText('Cancel'));
    await userEvent.click(screen.getByText('Confirm'));

    expect(mockHandleCancel).toBeCalled();
    expect(mockHandleConfirm).toBeCalled();

    done();
  })();
});
