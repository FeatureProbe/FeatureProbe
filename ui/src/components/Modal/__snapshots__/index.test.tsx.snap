// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot 1`] = `
<body
  class="dimmable dimmed"
>
  <div />
  <div
    class="ui page modals dimmer transition visible active"
    style="display: flex;"
  >
    <div
      class="ui mini modal transition visible active"
      style="width: 400px;"
    >
      <div
        class="content"
      >
        <div>
          modal content
        </div>
        <div
          class="footer"
        >
          <button
            class="ui mini basic button btn"
            type="reset"
          >
            Cancel
          </button>
          <button
            class="ui mini primary button"
            type="submit"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
